### PR TITLE
{2023.06}[foss/2021b] OpenFoam v2112

### PR DESCRIPTION
**Pull request by laraPPr**
edit (by @boegel, 2023-09-09):
```
60 out of 105 required modules missing:

* Bison/3.7.6-GCCcore-11.2.0 (Bison-3.7.6-GCCcore-11.2.0.eb)
* METIS/5.1.0-GCCcore-11.2.0 (METIS-5.1.0-GCCcore-11.2.0.eb)
* Eigen/3.4.0-GCCcore-11.2.0 (Eigen-3.4.0-GCCcore-11.2.0.eb)
* pkgconf/1.8.0-GCCcore-11.2.0 (pkgconf-1.8.0-GCCcore-11.2.0.eb)
* libpng/1.6.37-GCCcore-11.2.0 (libpng-1.6.37-GCCcore-11.2.0.eb)
* gzip/1.10-GCCcore-11.2.0 (gzip-1.10-GCCcore-11.2.0.eb)
* MPFR/4.1.0-GCCcore-11.2.0 (MPFR-4.1.0-GCCcore-11.2.0.eb)
* lz4/1.9.3-GCCcore-11.2.0 (lz4-1.9.3-GCCcore-11.2.0.eb)
* NASM/2.15.05-GCCcore-11.2.0 (NASM-2.15.05-GCCcore-11.2.0.eb)
* zstd/1.5.0-GCCcore-11.2.0 (zstd-1.5.0-GCCcore-11.2.0.eb)
* libjpeg-turbo/2.0.6-GCCcore-11.2.0 (libjpeg-turbo-2.0.6-GCCcore-11.2.0.eb)
* pixman/0.40.0-GCCcore-11.2.0 (pixman-0.40.0-GCCcore-11.2.0.eb)
* Brotli/1.0.9-GCCcore-11.2.0 (Brotli-1.0.9-GCCcore-11.2.0.eb)
* libiconv/1.16-GCCcore-11.2.0 (libiconv-1.16-GCCcore-11.2.0.eb)
* freetype/2.11.0-GCCcore-11.2.0 (freetype-2.11.0-GCCcore-11.2.0.eb)
* Doxygen/1.9.1-GCCcore-11.2.0 (Doxygen-1.9.1-GCCcore-11.2.0.eb)
* fontconfig/2.13.94-GCCcore-11.2.0 (fontconfig-2.13.94-GCCcore-11.2.0.eb)
* libgd/2.3.3-GCCcore-11.2.0 (libgd-2.3.3-GCCcore-11.2.0.eb)
* ICU/69.1-GCCcore-11.2.0 (ICU-69.1-GCCcore-11.2.0.eb)
* Boost/1.77.0-GCC-11.2.0 (Boost-1.77.0-GCC-11.2.0.eb)
* libcerf/1.17-GCCcore-11.2.0 (libcerf-1.17-GCCcore-11.2.0.eb)
* Ninja/1.10.2-GCCcore-11.2.0 (Ninja-1.10.2-GCCcore-11.2.0.eb)
* Meson/0.58.2-GCCcore-11.2.0 (Meson-0.58.2-GCCcore-11.2.0.eb)
* Mako/1.1.4-GCCcore-11.2.0 (Mako-1.1.4-GCCcore-11.2.0.eb)
* Python/2.7.18-GCCcore-11.2.0-bare (Python-2.7.18-GCCcore-11.2.0-bare.eb)
* ffnvcodec/11.1.5.2 (ffnvcodec-11.1.5.2.eb)
* PCRE/8.45-GCCcore-11.2.0 (PCRE-8.45-GCCcore-11.2.0.eb)
* GLib/2.69.1-GCCcore-11.2.0 (GLib-2.69.1-GCCcore-11.2.0.eb)
* X11/20210802-GCCcore-11.2.0 (X11-20210802-GCCcore-11.2.0.eb)
* cairo/1.16.0-GCCcore-11.2.0 (cairo-1.16.0-GCCcore-11.2.0.eb)
* GObject-Introspection/1.68.0-GCCcore-11.2.0 (GObject-Introspection-1.68.0-GCCcore-11.2.0.eb)
* netCDF/4.8.1-gompi-2021b (netCDF-4.8.1-gompi-2021b.eb)
* SCOTCH/6.1.2-gompi-2021b (SCOTCH-6.1.2-gompi-2021b.eb)
* libdrm/2.4.107-GCCcore-11.2.0 (libdrm-2.4.107-GCCcore-11.2.0.eb)
* re2c/2.2-GCCcore-11.2.0 (re2c-2.2-GCCcore-11.2.0.eb)
* HarfBuzz/2.8.2-GCCcore-11.2.0 (HarfBuzz-2.8.2-GCCcore-11.2.0.eb)
* x264/20210613-GCCcore-11.2.0 (x264-20210613-GCCcore-11.2.0.eb)
* libglvnd/1.3.3-GCCcore-11.2.0 (libglvnd-1.3.3-GCCcore-11.2.0.eb)
* double-conversion/3.1.5-GCCcore-11.2.0 (double-conversion-3.1.5-GCCcore-11.2.0.eb)
* FriBidi/1.0.10-GCCcore-11.2.0 (FriBidi-1.0.10-GCCcore-11.2.0.eb)
* LAME/3.100-GCCcore-11.2.0 (LAME-3.100-GCCcore-11.2.0.eb)
* Pango/1.48.8-GCCcore-11.2.0 (Pango-1.48.8-GCCcore-11.2.0.eb)
* libunwind/1.5.0-GCCcore-11.2.0 (libunwind-1.5.0-GCCcore-11.2.0.eb)
* PCRE2/10.37-GCCcore-11.2.0 (PCRE2-10.37-GCCcore-11.2.0.eb)
* LLVM/12.0.1-GCCcore-11.2.0 (LLVM-12.0.1-GCCcore-11.2.0.eb)
* graphite2/1.3.14-GCCcore-11.2.0 (graphite2-1.3.14-GCCcore-11.2.0.eb)
* Yasm/1.3.0-GCCcore-11.2.0 (Yasm-1.3.0-GCCcore-11.2.0.eb)
* Mesa/21.1.7-GCCcore-11.2.0 (Mesa-21.1.7-GCCcore-11.2.0.eb)
* libGLU/9.0.2-GCCcore-11.2.0 (libGLU-9.0.2-GCCcore-11.2.0.eb)
* x265/3.5-GCCcore-11.2.0 (x265-3.5-GCCcore-11.2.0.eb)
* FFmpeg/4.3.2-GCCcore-11.2.0 (FFmpeg-4.3.2-GCCcore-11.2.0.eb)
* snappy/1.1.9-GCCcore-11.2.0 (snappy-1.1.9-GCCcore-11.2.0.eb)
* NSPR/4.32-GCCcore-11.2.0 (NSPR-4.32-GCCcore-11.2.0.eb)
* NSS/3.69-GCCcore-11.2.0 (NSS-3.69-GCCcore-11.2.0.eb)
* JasPer/2.0.33-GCCcore-11.2.0 (JasPer-2.0.33-GCCcore-11.2.0.eb)
* Qt5/5.15.2-GCCcore-11.2.0 (Qt5-5.15.2-GCCcore-11.2.0.eb)
* CGAL/4.14.3-gompi-2021b (CGAL-4.14.3-gompi-2021b.eb)
* ParaView/5.9.1-foss-2021b-mpi (ParaView-5.9.1-foss-2021b-mpi.eb)
* gnuplot/5.4.2-GCCcore-11.2.0 (gnuplot-5.4.2-GCCcore-11.2.0.eb)
* OpenFOAM/v2112-foss-2021b (OpenFOAM-v2112-foss-2021b.eb)
```

### CODE REVIEW

**Comment by laraPPr** [comment]
bot: build repo:eessi-2023.06-software

**Comment by laraPPr** [comment]
`7216`
```
== 2023-09-08 08:40:15,718 build_log.py:171 ERROR EasyBuild crashed with an error (at easybuild/tools/build_log.py:111 in caller_info): Build of /cvmfs/pilot.eessi-hpc.org/versions/2023.06/software/linux/x86_64/generic/software/EasyBuild/4.8.0/easybuild/easyconfigs/m/MPFR/MPFR-4.1.0-GCCcore-11.2.0.eb failed (err: 'build failed (first 300 chars): cmd "mkdir -p /cvmfs/pilot.eessi-hpc.org/versions/2023.06/software/linux/x86_64/generic/software/MPFR/4.1.0-GCCcore-11.2.0/lib && cp -a src/.libs/libmpfr.so* /cvmfs/pilot.eessi-hpc.org/versions/2023.06/software/linux/x86_64/generic/software/MPFR/4.1.0-GCCcore-11.2.0/lib &&  make check  && rm -r /cvm') (at easybuild/main.py:174 in build_and_install_software)
```

**Comment by boegel** [comment]
Hmm, looks like it's because a single test failed:

```
...
FAIL: tsprintf
...
============================================================================
Testsuite summary for MPFR 4.1.0
============================================================================
# TOTAL: 183
# PASS:  182
# SKIP:  0
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0
```

**Comment by boegel** [comment]
@laraPPr Can you check if you find any clues upstream on this?

```
mpfr-4.1.0/tests/test-suite.log
======================================
   MPFR 4.1.0: tests/test-suite.log
======================================

# TOTAL: 183
# PASS:  182
# SKIP:  0
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0

.. contents:: :depth: 2

FAIL: tsprintf
==============

Error in mpfr_vsprintf (s, "%0+ -'13.10Pd:", ...);
expected: "+01,234,567  :"
got:      "+0001,234,567:"
FAIL tsprintf (exit status: 1)
```

**Comment by laraPPr** [comment]
I found this, [here](https://www.mpfr.org/mpfr-4.2.0/#fixed), but I don't know if it is relevant to us:

A test of the thousands separator in tsprintf.c is based on the output from the GNU C Library up to 2.36, which is [incorrect](https://sourceware.org/bugzilla/show_bug.cgi?id=23432). The output has changed in 2.37 (partly fixed), so that tsprintf fails with glibc 2.37. The [tsprintf-thousands patch](https://www.mpfr.org/mpfr-4.2.0/patch01) modifies the test to conform to POSIX and also avoid the buggy case in 2.36 and below. However, this new test, which was expected to succeed, triggers a serious bug in 2.37 ([bug 30068 / CVE-2023-25139](https://sourceware.org/bugzilla/show_bug.cgi?id=30068)). We did not modify the test again since this bug affects MPFR's mpfr_sprintf function, with a possible buffer overflow in particular cases. This bug has been [fixed in the 2.37 branch](https://sourceware.org/git/?p=glibc.git;a=commit;h=07b9521fc6369d000216b96562ff7c0ed32a16c4). In short, this patch is useful (and needed) for a fixed glibc 2.37 and some other libraries, depending on the current locales.
Corresponding changesets in the 4.2 branch: [4f03d40b5](https://gitlab.inria.fr/mpfr/mpfr/-/commit/4f03d40b5a5ceeb3126dc51c3511bdceef3bab19), [78ff7526d](https://gitlab.inria.fr/mpfr/mpfr/-/commit/78ff7526dea103f73acc98d9a9ffa7429050c0dd), [e66bb7121](https://gitlab.inria.fr/mpfr/mpfr/-/commit/e66bb71212b2789d783ccbc3b5b186247b972d5b).

**Comment by boegel** [comment]
OK, excellent, so we should try and apply that patch to MPFR in EasyBuild, and then use `--from-pr` here.

**Comment by boegel** [comment]
@laraPPr Let's see if https://github.com/easybuilders/easybuild-easyconfigs/pull/18746 fixes the MPFR test problem

**Comment by laraPPr** [comment]
bot: build repo:eessi-2023.06-software arch:x86_64/generic

**Comment by laraPPr** [comment]
bot: build repo:eessi-2023.06-software arch:x86_64/generic

**Comment by laraPPr** [comment]
bot: build repo:eessi-2023.06-software arch:x86_64/intel/haswell 
bot: build repo:eessi-2023.06-software arch:x86_64/intel/skylake_avx512
bot: build repo:eessi-2023.06-software arch:x86_64/amd/zen2
bot: build repo:eessi-2023.06-software arch:x86_64/amd/zen3
bot: build repo:eessi-2023.06-software arch:aarch64/generic
bot: build repo:eessi-2023.06-software arch:aarch64/neoverse_n1
bot: build repo:eessi-2023.06-software arch:aarch64/neoverse_v1

**Comment by boegel** [comment]
All 8 staging PRs merged, deployment under way...

File: eessi-2023.06-eb-4.8.0-2021b.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+easyconfigs:
+  - MPFR-4.1.0-GCCcore-11.2.0.eb:
+      #fix for different output of tsprintf.c test from GNU C library <2.36
+      #see https://github.com/easybuilders/easybuild-easyconfigs/pull/18746
```
Comment (boegel) [inline eessi-2023.06-eb-4.8.0-2021b.yml:4]:
```suggestion
      # use patch for MPFR 4.1.0 to fix failing tsprintf test with glibc >= 2.37,
      # see https://github.com/easybuilders/easybuild-easyconfigs/pull/18746

**Review by boegel — CHANGES_REQUESTED** [review]

**Review by boegel — APPROVED** [review]

**Pull request merged**